tests/ui/asm/x86_64/issue-96797.rs

```diff
@@ -11,7 +11,14 @@ use std::arch::global_asm;
 #[no_mangle]
 fn my_func() {}
 
-global_asm!("call_foobar: jmp {}", sym foobar);
+global_asm!("
+.globl call_foobar
+.type call_foobar,@function
+.section .text.call_foobar,\"ax\",@progbits
+call_foobar: jmp {}
+.size call_foobar, .-call_foobar
+.text
```

Comment:
Should this instead be using `.pushsection` and `.popsection` to make sure that the section is not modified by starting in something other than plain `.text`

Reply:
Done